fe2-android/app/src/main/java/com/github/dedis/popstellar/model/qrcode/ConnectToLao.java

```diff
@@ -0,0 +1,21 @@
+package com.github.dedis.popstellar.model.qrcode;
+
+import com.github.dedis.popstellar.model.network.serializer.JsonUtils;
+import com.google.android.gms.vision.barcode.Barcode;
+import com.google.gson.Gson;
+
+public class ConnectToLao {
+
+  public final String server;
+  public final String lao;
+
+  public ConnectToLao(String server, String lao) {
+    this.server = server;
+    this.lao = lao;
+  }
+
+  public static ConnectToLao extractFrom(Gson mGson, Barcode barcode) {
```

Comment:
do we need the `Barcode` dependency here, or could we just pass in a string containing `barcode.rawValue` ? 
it seems like we could avoid coupling `ConnectToLao` with `Barcode`

Reply:
Yes, but it could also be changed in `onQRCodeDetected`. I will change that.

And I tested it locally from android to android. I was just not able to to it under automated tests.